src/test/java/org/jmxtrans/agent/influxdb/InfluxDbOutputWriterTest.java

```diff
@@ -119,4 +119,39 @@ public void manyMetrics() throws Exception {
                 .withRequestBody(equalTo("foo value=1i 1234\nfoo2 value=2.0 1234")));
     }
 
+    @Test
+    public void requestWhenDisabled() throws Exception {
+        Map<String, String> s = new HashMap<>();
+        s.put("url", "http://localhost:" + wireMockRule.port());
+        s.put("database", "test-db");
+        s.put("enabled", "false");
+        // NOTE: This is a workaround as I didn't see a clean method to assert that Wiremocks is not called.
+        stubFor(any(urlPathEqualTo("/write"))
+                .atPriority(10)
+                .willReturn(aResponse()
+                        .withStatus(404)
+                        .withBody("{\"status\":\"Error\",\"message\":\"Should not be called\"}")));
+        InfluxDbOutputWriter writer = new InfluxDbOutputWriter(FAKE_CLOCK);
+        writer.postConstruct(s);
+        writer.writeQueryResult("foo", null, 1);
+        writer.postCollect();
+    }

```

Comment:
There should be a verification that wire mock has not been called instead on relying on the HTTP 400 to throw an exception. Which also means that the declaration of the stub could be much simpler (as it *should* never be called, no need to specify a complex stub).